templates/core/terraform/firewall/firewall.tf

```diff
@@ -23,16 +23,19 @@ resource "azurerm_firewall" "fw" {
 }
 
 resource "azurerm_management_lock" "fw" {
+  count      = var.debug ? 0 : 1
   name       = azurerm_firewall.fw.name
   scope      = azurerm_firewall.fw.id
   lock_level = "CanNotDelete"
   notes      = "Locked to prevent accidental deletion"
 }
 
 resource "azurerm_monitor_diagnostic_setting" "firewall" {
-  name                       = "diagnostics-firewall-${var.tre_id}"
-  target_resource_id         = azurerm_firewall.fw.id
-  log_analytics_workspace_id = var.log_analytics_workspace_id
+  name                           = "diagnostics-firewall-${var.tre_id}"
+  target_resource_id             = azurerm_firewall.fw.id
+  log_analytics_workspace_id     = var.log_analytics_workspace_id
+  log_analytics_destination_type = "Dedicated"
```

Comment:
It wasn't clear how this impacts start/stop stuff.
Is this part of that change, or unrelated?

Reply:
Kind of related:
If _log_analytics_destination_type_ isn't set then TF uses a default value of _AzureDiagnostics_. After the deployment that value somehow changes into _null_ (probably something on Azure).
On some runs, TF tries to recreate this resource and fails (something about the resource already there). I saw that a few times while testing the stop/start. 
The easy solution was to make sure TF doesn't try to change it at all. A value of _Dedicated_ is the way forward anyway so it made sense.